Use istio operator for installing istio (#1382)
kyma-incubator/reconciler: pkg/reconciler/instances/istio/README.md

## Overview

> Istio Reconciler manages the [Istio](https://github.com/kyma-project/kyma/tree/main/resources/istio) Kyma component. The component in Kyma provides definition of [which version of Istio module should be installed](https://github.com/kyma-project/kyma/blob/main/resources/istio/templates/istio-manager-config.yaml).

Reviewer: ```suggestion
Istio Reconciler manages the [Istio](https://github.com/kyma-project/kyma/tree/main/resources/istio) Kyma component. The component in Kyma defines [which version of the Istio module should be installed](https://github.com/kyma-project/kyma/blob/main/resources/istio/templates/istio-manager-config.yaml).
```

   make build-darwin
   ```

3. Run Istio Reconciler:

   ```bash
    ./bin/mothership-darwin local --components istio
   ```

## Details

Reconciliation in Kyma is handled by Reconciler. The Mothership Reconciler knows the reconciliation status of every managed Kyma cluster and initiates reconciliation of all Kyma components.

> Istio component reconciler resposibility is installation and upgrades of [Istio Module](https://github.com/kyma-project/istio) on the cluster and keeping the state of Istio Custom Resource Ready.

Reviewer: ```suggestion
The Istio component's reconciler is responsible for installing and upgrading the [Istio Module](https://github.com/kyma-project/istio) on the cluster. It also ensures that the Istio custom resource remains in the `Ready` state.
```

## Istio reconciler workflow

> 1. Istio reconciler fetches the version of to-be-installed istio module from Kyma charts definition under `resources/istio/templates/istio-manager-config.yaml`

Reviewer: ```suggestion
1. Istio reconciler fetches the version of the Istio module that needs to be installed from the Kyma charts definition located in `resources/istio/templates/istio-manager-config.yaml`.
```

> 2. Istio module installation manifests and default Istio CR is fetched from [Istio module release](https://github.com/kyma-project/istio/releases)

Reviewer: ```suggestion
2. The Istio module's installation manifests and the default Istio CR are fetched from the [Istio module release](https://github.com/kyma-project/istio/releases).
```

> 3. The module installation charts are installed or updated

Reviewer: ```suggestion
3. The module's installation charts are either installed or updated.
```

> 4. If there is no Istio Custom Resource present, reconciler creates a new one, triggering istio-manager reconcilation

Reviewer: ```suggestion
4. If there is no Istio CR present, the reconciler creates a new one. This triggers the reconciliation of `istio-manager`.
```

> 5. Reconciler wait's until Istio module is Ready, then finishes reconciling

Reviewer: ```suggestion
5. The reconciler waits until the Istio module is in the `Ready` state and then finishes the reconciliation process.
```
